### PR TITLE
Fix max connections in DruidAvaticaHandlerTest

### LINKED ISSUE #5183: DruidAvaticaHandlerTest transient failure (times out)

### PR DESCRIPTION
**Pull request by jon-wei**
Fixes #5183 

DruidAvaticaHandlerTest uses 3 Connections (client, superuserClient, and clientLosAngeles) but the connection limit was set at 2.

This was causing testConcurrentQueries to fail sometimes when `client` (the connection object used by the test) was cleared from the connection map, and the multiple concurrent threads attempted to re-open the connection

### CODE REVIEW

**Comment by jon-wei** [comment]
perhaps `openDruidConnection` in `DruidMeta` needs to be adjusted re: concurrency, but that requires some more thought and this PR is just meant as a quick fix to make the test stop failing

File: sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java [hunk 1]
```diff
@@ -96,7 +96,7 @@
     @Override
     public int getMaxConnections()
     {
-      return 2;
+      return 3;
```
Comment (leventov) [inline sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java]: Could you leave a comment, or extract a constant, or somehow else prevent regression in the future?

Comment (jon-wei) [reply sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java]: added a comment

**Comment by SmaatoDataTeam** [comment]
facing same issue of verion 0.20.1

**Pull request merged**